set-java-home.bash

```diff
@@ -0,0 +1,14 @@
+asdf_update_java_home() {
+  local java_path
+  java_path="$(asdf which java)"
+  if [[ ! -z "${java_path}" ]]; then
+    export JAVA_HOME
+   JAVA_HOME="$(dirname "$(dirname "$(realpath "${java_path}")")")"
+  fi
+}
+
+prompt_command() {
+  asdf_update_java_home
```

Comment:
I would add here some mention about any preexisting `PROMPT_COMMAND`, since we will end up replacing it with the `export` in next lines.

In GNOME-like terminals the system already does this trick, and calls a function `__vte_prompt_command`. In those systems you have to keep calling that function.

My real setup looks like:

```bash
function asdf_update_java_home {
  asdf current java 2>&1 > /dev/null
  if [[ "$?" -eq 0 ]]
  then
      export JAVA_HOME=$(asdf where java)
  fi
}

# auto activate virtualenv when entering its root path
function auto_activate_virtualenv {
  if [[ "$VIRTUAL_ENV" = "" && -f "$PWD/env/bin/activate" ]]; then
    source "$PWD/env/bin/activate"
  fi
}

function prompt_command {
  __vte_prompt_command
  asdf_update_java_home
  auto_activate_virtualenv
}

export PROMPT_COMMAND=prompt_command
```